diff --git a/eternal-darkness-decomp/src/game/game_fn_80191034.c b/eternal-darkness-decomp/src/game/game_fn_80191034.c
index ddf500924..6fee872aa 100644
--- a/eternal-darkness-decomp/src/game/game_fn_80191034.c
+++ b/eternal-darkness-decomp/src/game/game_fn_80191034.c
@@ -15,8 +15,9 @@ extern int fn_801ED57C(int);
 extern void fn_8018D0D0(void*, void*, s16);
 extern void fn_80188868(void*, void*, void*, float);
 
-void fn_80191034(u8* data)
+void fn_80191034(void* arg)
 {
+    u8* data = arg;
     u8* table = lbl_80607120;
     int i;
     u8 count = data[1];
