--- baseline
+++ initialized-long-inner-edges
@@ -14,10 +14,10 @@
     s16 right = (s16)x + width;
     s16 bottom = (s16)y + height;
     s16 shortInset;
-    s32 topInset;
+    s32 topInset = (s16)y;
     s32 rightInset;
     s32 bottomInset;
-    s32 leftInset;
+    s32 leftInset = (s16)x;
     u32 copy = *color;
 
     fn_801A852C(&copy, 0, -1, 0x80000000);
@@ -26,7 +26,7 @@
     fn_801A9454((s16)x, (s16)y, (s16)depth);
     fn_801A9454(right, (s16)y, (s16)depth);
     shortInset = inset;
-    topInset = (s16)y + shortInset;
+    topInset += shortInset;
     fn_801A9454(right, (s16)topInset, (s16)depth);
     fn_801A9454((s16)x, (s16)topInset, (s16)depth);
     rightInset = right - shortInset;
@@ -40,7 +40,7 @@
     fn_801A9454(right, bottom, (s16)depth);
     fn_801A9454((s16)x, bottom, (s16)depth);
     fn_801A9454((s16)x, (s16)topInset, (s16)depth);
-    leftInset = (s16)x + shortInset;
+    leftInset += shortInset;
     fn_801A9454((s16)leftInset, (s16)topInset, (s16)depth);
     fn_801A9454((s16)leftInset, (s16)bottomInset, (s16)depth);
     fn_801A9454((s16)x, (s16)bottomInset, (s16)depth);
