--- baseline
+++ unsigned-short-inset-locals
@@ -14,10 +14,10 @@
     s16 right = (s16)x + width;
     s16 bottom = (s16)y + height;
     s16 shortInset;
-    s32 topInset;
-    s32 rightInset;
-    s32 bottomInset;
-    s32 leftInset;
+    unsigned short topInset;
+    unsigned short rightInset;
+    unsigned short bottomInset;
+    unsigned short leftInset;
     u32 copy = *color;
 
     fn_801A852C(&copy, 0, -1, 0x80000000);
